pkg/kube/cluster-init.sh

```diff
@@ -283,18 +283,30 @@ apply_node_uuid_lable () {
 
 reapply_node_labes() {
         apply_node_uuid_lable
-        apply_longhorn_disk_config() "$HOSTNAME"
+        apply_longhorn_disk_config "$HOSTNAME"
         # Check if the node with both labels exists, don't assume above apply worked
         node_count=$(kubectl get nodes -l node-uuid="$DEVUUID",node.longhorn.io/create-default-disk=config -o json | jq '.items | length')
 
         if [ "$node_count" -gt 0 ]; then
                 logmsg "Node labels re-applied successfully"
                 touch /var/lib/node-labels-initialized
         else
-                logmsg "Failed to re-apply node labels"
+                logmsg "Failed to re-apply node labels, on $HOSTNAME, uuid $DEVUUID"
         fi
 }
 
+# when transitioning from single node to cluster mode, the k3s.yaml file may need
+# to change with new certificates
+check_kubeconfig_yaml_files() {
```

Comment:
Do we really need to cmp and copy. Why can't we blindly copy whenever we switch to cluster mode ?

Reply:
No this is not during the swtichover time, when switch from single node to cluster, it actually generated the k3s.yaml twice, one is initially which we does the copy as normal, then after joined the cluster, it gets it from the bootstrap server. So, we don't know when this is happening (well, within some 30 seconds).